qualtran/conftest.py

```diff
@@ -97,6 +97,9 @@ def assert_bloq_example_serializes_for_pytest(bloq_ex: BloqExample):
         'tensor_product_block_encoding',
         'tensor_product_block_encoding_override',
         'tensor_product_block_encoding_symb',
+        'product_block_encoding',
+        'product_block_encoding_override',
+        'product_block_encoding_symb',
```

Comment:
Why are these not serializable ?

Reply:
@charlesyuan314 I'll merge the PR and we can continue this discussion and potentially address in a follow-up PR

Reply:
I believe it is because this bloq has an attribute of type `Tuple[BlockEncoding, ...]`. It results in the following at serialization time:
```
qualtran.testing.BloqCheckException: Serialization Failed:
Object arrays cannot be saved when allow_pickle=False
```

Are bloqs contained as attributes something that can be made serializable?

Reply:
> Object arrays cannot be saved when allow_pickle=False

I think it's because so far we haven't had any bloq that stores a sequence of subbloqs as an attribute. 

To support serialization we'll have to update https://github.com/quantumlib/Qualtran/blob/f8d2db04b2986268fbe423f0e294b10152758736/qualtran/protos/bloq.proto#L40 to add a new ` NDArray subbloqs`  that stores an NDArray of integers corresponding to IDs of each of the subbloqs and then update https://github.com/quantumlib/Qualtran/blob/f8d2db04b2986268fbe423f0e294b10152758736/qualtran/serialization/bloq.py#L349 to check if the argument is an ndarray of bloqs, then delegate to a different method that serializes it. 

Can you please open an issue to track this?